src/lib/RoutingManager.ts

```diff
@@ -35,139 +34,44 @@ class RoutingManager implements Widget {
     this.router = router;
     this.stateMapping = stateMapping;
     this.instantSearchInstance = instantSearchInstance;
-    this.indexId = this.instantSearchInstance.indexName;
     this.currentUiState = this.stateMapping.routeToState(this.router.read());
-  }
 
-  private getAllSearchParameters({
-    currentSearchParameters,
-    uiState,
-  }: {
-    currentSearchParameters: SearchParameters;
-    uiState: UiState;
-  }): SearchParameters {
-    const widgets = this.instantSearchInstance.mainIndex.getWidgets();
-    const indexUiState = uiState[this.indexId] || {};
-
-    return widgets.reduce((parameters, widget) => {
-      if (!widget.getWidgetSearchParameters) {
-        return parameters;
-      }
-
-      return widget.getWidgetSearchParameters(parameters, {
-        uiState: indexUiState,
-      });
-    }, currentSearchParameters);
+    this.createURL = this.createURL.bind(this);
   }
 
-  private getAllUiStates({
-    searchParameters,
-  }: {
-    searchParameters: SearchParameters;
-  }): UiState {
-    const widgets = this.instantSearchInstance.mainIndex.getWidgets();
-    const helper = this.instantSearchInstance.mainIndex.getHelper()!;
-
-    return {
-      [this.indexId]: widgets.reduce<IndexUiState>((state, widget) => {
-        if (!widget.getWidgetState) {
-          return state;
+  public setupRouting(): void {
```

Comment:
what about `applyDefaultParameters`? It's supposed to be set on initialisation of InstantSearch only, and it's there to apply the search parameters from the URL. If you have another idea that's also cool :)

Reply:
What about `applyStateFromRoute`?

Reply:
sounds good too for me! It's of course a little confusing because route isn't passed as an argument, but good name

Reply:
Agree, since route is passed to the constructor it is kind of OK I guess.